Assignment_1/palindrome_permutation.js

```diff
@@ -0,0 +1,33 @@
+const palindromePermutation = (str) => {
+
+    const charMap = new Map(); //frequency of each char in string
+    str = str.replace(/\s/g, '').toLowerCase();
+    //remove spaces and convert string to lower case
+  //
+    for (const char of str) {
+        if (charMap.has(char)) {
+          charMap.set(char, charMap.get(char) + 1);
+        } else {
+          charMap.set(char, 1);
+        }
+      }
+
+    //checking palindrome
+    let count = 0;
+    for (const i of charMap.values()) {
+        if (i % 2 !== 0) {
+        count++;
+        }
+    }
+    //if one characater with odd freq
+    return count <=1;
```

Comment:
Although there's nothing wrong with the way you did this, this would be a good place to use JS's `Array.prototype.filter` method. Because this method only works on arrays, you can use the spread operator (`...`) to coerce `charMap.values()` from an iterator to an array, and then filter so that you're only left with a list containing the odd values.
```suggestion
    return [...charMap.values()].filter(value => value % 2 !== 0).length <= 1;
```